pkg/feature/api/feature.go

```diff
@@ -660,7 +660,107 @@ func (s *FeatureService) UpdateFeature(
 	ctx context.Context,
 	req *featureproto.UpdateFeatureRequest,
 ) (*featureproto.UpdateFeatureResponse, error) {
-	return nil, status.Error(codes.Unimplemented, "method UpdateFeatureDetails not implemented")
+	localizer := locale.NewLocalizer(ctx)
+	editor, err := s.checkEnvironmentRole(
+		ctx, accountproto.AccountV2_Role_Environment_EDITOR,
+		req.EnvironmentId, localizer)
+	if err != nil {
+		return nil, err
+	}
+	if req.Id == "" {
+		dt, err := statusMissingID.WithDetails(&errdetails.LocalizedMessage{
+			Locale:  localizer.GetLocale(),
+			Message: localizer.MustLocalizeWithTemplate(locale.RequiredFieldTemplate, "id"),
+		})
+		if err != nil {
+			return nil, statusInternal.Err()
+		}
+		return nil, dt.Err()
+	}
+	if err := s.validateFeatureStatus(ctx, req.Id, req.EnvironmentId, localizer); err != nil {
+		return nil, err
+	}
+	if err := s.validateEnvironmentSettings(ctx, req.EnvironmentId, req.Comment, localizer); err != nil {
+		return nil, err
+	}
+	var handler *command.FeatureCommandHandler = command.NewEmptyFeatureCommandHandler()
+	tx, err := s.mysqlClient.BeginTx(ctx)
+	if err != nil {
+		s.logger.Error(
+			"Failed to begin transaction",
+			log.FieldsFromImcomingContext(ctx).AddFields(
+				zap.Error(err),
+			)...,
+		)
+		dt, err := statusInternal.WithDetails(&errdetails.LocalizedMessage{
+			Locale:  localizer.GetLocale(),
+			Message: localizer.MustLocalize(locale.InternalServerError),
+		})
+		if err != nil {
+			return nil, statusInternal.Err()
+		}
+		return nil, dt.Err()
+	}
+	err = s.mysqlClient.RunInTransaction(ctx, tx, func() error {
+		featureStorage := v2fs.NewFeatureStorage(tx)
+		feature, err := featureStorage.GetFeature(ctx, req.Id, req.EnvironmentId)
+		if err != nil {
+			s.logger.Error(
+				"Failed to get feature",
+				log.FieldsFromImcomingContext(ctx).AddFields(
+					zap.Error(err),
+					zap.String("environmentId", req.EnvironmentId),
+				)...,
+			)
+			return err
+		}
+		handler = command.NewFeatureCommandHandler(editor, feature, req.EnvironmentId, req.Comment)
```

Comment:
You commented that we won't use commands for new APIs used by the public API.
What's the reason for using it here?

If you are going to use commands, why not use the current APIs instead of creating new ones using the same code?